packages/react/src/components/Table/tableReducer.js

```diff
@@ -212,17 +222,46 @@ export const filterSearchAndSort = (
   columns,
   advancedFilters = []
 ) => {
-  const { columnId, direction } = sort;
-
   const { value: searchValue } = search;
   const filteredData = filterData(data, filters, columns, advancedFilters);
   const searchedData =
     searchValue && searchValue !== '' ? searchData(filteredData, searchValue) : filteredData;
-  return !isEmpty(sort)
-    ? getCustomColumnSort(columns, columnId)
-      ? getCustomColumnSort(columns, columnId)({ data: searchedData, columnId, direction })
-      : getSortedData(searchedData, columnId, direction)
-    : searchedData;
+
+  if (isEmpty(sort)) {
+    return searchedData;
+  }
+
+  if (Array.isArray(sort)) {
+    let sortStack = firstBy(() => 0);
+    sort.forEach(({ columnId, direction }) => {
+      const customSortFn = getCustomColumnSort(columns, columnId);
+      if (customSortFn) {
+        const sortedValues = customSortFn({ data: searchedData, columnId, direction }).map(
```

Comment:
I think it's worth discussing a breaking change in V3 here, so that the customSort function is just a comparator instead of a method that takes the data, column, and direction. It would be much easier (and more efficient) to handle sorting related issues without having to maintain this backwards compatibility. Unless this method was chosen for a specific reason that I'm unaware of.

Reply:
I think it was a functions so you can sort things that are not comparable like an error icon.

Reply:
🤔  Wouldn't that still be possible with a comparator? They can use the `renderDataFunction` to render the icon, and then `sortFunction` could still just be a [comparator as used here](https://github.com/carbon-design-system/carbon-addons-iot-react/blob/next/packages/react/src/components/Table/Table.story.jsx#L106). It seems like the only thing the full function is accomplishing to me is cloning to not mutate the original array.